### ISSUE TITLE
Some files are ignored

### ISSUE THREAD

**Issue by NZhuravlev** [webpack-contrib/eslint-webpack-plugin#120]
- Operating System:  macOS Big Sur 11.5.2
- Node Version: v14.17.6
- NPM Version: 6.14.15
- webpack Version: 4.46.0
- eslint-webpack-plugin Version: 2.5.4

### Expected Behavior

Errors in all the specified folders should be reported.

### Actual Behavior

With either

```
new ESLintPlugin({
	files: ["src", "api-examples"],
	extensions: [".ts"],
	emitWarning: true,
	emitError: true,
	failOnWarning: true,
}),
```

or 

```
new ESLintPlugin({
	extensions: [".ts"],
	emitWarning: true,
	emitError: true,
	failOnWarning: true,
}),
```

errors in the api-examples (one of the folders in the project) are not getting reported.

### Code

eslintignore:

```
npm-api/*
```

.eslintrc.js:

```
module.exports = {
	"parser": "@typescript-eslint/parser",
	"parserOptions": {
		"ecmaVersion": 6,
		"sourceType": "module",
		"project": "./tsconfig.eslint.json",
		"tsconfigRootDir": __dirname
	},
...
```

tsconfig.eslint.json:

```
{
        ...
	"include": [
		"src/**/*.ts",
		"api-examples/**/*.ts"
	],
	"exclude": [
		"npm-api"
	]
}
```

### How Do We Reproduce?

If I were to guess it's about setting up a ts project with similar rules. api-examples folder files here are not used by webpack while compiling - it might be a problem. The documentation says it's possible to lint files outside the compiler context.

**Comment by riteshjagga** [comment]
It would be nice to have them in the scope of the plugin. We have a case where our spec files are in different directory than the src files but would like to have spec files follow consistent styling too.

Eslint can be directly used to lint multiple directories but that would mean that we'll need to connect the eslint failures to fail the webpack compilation too somehow, which I do not know how to do it, at the time of writing.

This plugin brings the benefit of failing the webpack compilation (which was one of the big reason it was chosen for our project) in case of any linting errors and developers are then enforced by the means of errors to fix them which brings consistent styling.
